pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -125,3 +125,41 @@ func (s *REST) findStreamForMapping(ctx kapi.Context, mapping *api.ImageStreamMa
 	}
 	return nil, errors.NewNotFound("ImageStream", "")
 }
+
+// Delete deletes the given image and the image stream tag associated with it
```

Comment:
Do not delete the image. Only delete the tag. `oadm prune images` will delete the image, if appropriate.